src/Mod/Start/Gui/FileCardDelegate.cpp

```diff
@@ -63,23 +63,30 @@ void FileCardDelegate::paint(QPainter* painter,
     painter->save();
 
     // Step 1: Styling
-    QStyleOptionButton buttonOption;
-    buttonOption.initFrom(_widget.get());
-    buttonOption.rect = option.rect;
-    buttonOption.state = QStyle::State_Enabled;
-
-    if ((option.state & QStyle::State_MouseOver) != 0) {
-        buttonOption.state |= QStyle::State_MouseOver;
-    }
-    if ((option.state & QStyle::State_Selected) != 0) {
-        buttonOption.state |= QStyle::State_On;
+    if (qApp->styleSheet().isEmpty()) {
+        QStyleOptionButton buttonOption;
+        buttonOption.initFrom(_widget.get());
+        buttonOption.rect = option.rect;
+        buttonOption.state = QStyle::State_Enabled;
+        if ((option.state & QStyle::State_MouseOver) != 0) {
+            buttonOption.state |= QStyle::State_MouseOver;
+        }
+        if ((option.state & QStyle::State_Selected) != 0) {
+            buttonOption.state |= QStyle::State_On;
+        }
+        if ((option.state & QStyle::State_Sunken) != 0) {
+            buttonOption.state |= QStyle::State_Sunken;
+        }
+        QApplication::style()->drawControl(QStyle::CE_PushButton, &buttonOption, painter);
     }
-    if ((option.state & QStyle::State_Sunken) != 0) {
-        buttonOption.state |= QStyle::State_Sunken;
+    else {
+        QPushButton button;
+        button.resize(option.rect.size());
+        painter->translate(option.rect.topLeft());
+        button.render(painter);
+        painter->translate(-option.rect.topLeft());

```

Comment:
Generally it looks fine however I'm not sure if instantiating widgets in paint which should be pretty fast is a good idea. Do you think that we have a better way of achieving the same result? For example preinstantiate the widget and reuse it between paints?